### DISCUSSION TITLE
Page Not Found on Vercel App buts Works Locally

### DISCUSSION THREAD

**Discussion by tanu-chahal** [vercel/community#3780] [Help] [answered]
### Summary

Hey Vercel Community!

I just deployed my app on Vercel it got deployed properly with no build errors, and when I click on the URL it does display the home page.

It also displays the login page & the signup page.

But my app has one more page with the route (/games?reload=true) which works locally, but on the  Vercel app it says 404:NOT_FOUND.

If anyone can provide some help please do, will genuinely appreciate?

On Vercel App
![Screenshot (101)](https://github.com/vercel/community/assets/117712672/4bc3583d-fd65-4742-a97c-e330c238893c)

Locally
![Screenshot (102)](https://github.com/vercel/community/assets/117712672/b12d75af-c6de-4f13-bf5a-f61233105d79)

### Steps to Reproduce

Visit https://chessbuddy.vercel.app/

Login as a temp user 
email: tempuser@gmail.com
password: tempuser

after successfully logging in click on "Play" button to go on /games page

There it shows 404:NOT_FOUND error.
![Screenshot (101)](https://github.com/vercel/community/assets/117712672/596daa16-6c81-4460-838c-10ecbf0cc7bb)

**Comment by amyegan** [comment]
Hey @tanu-chahal. When trying to log in, the page didn't do anything for a while, and then I was redirected to a 404 error page. I was not able to get to the point where I could click "play" to go reach the `/games` page. 

Do you see more information in the [runtime logs](https://vercel.com/docs/observability/runtime-logs) for this deployment?

**Reply by tanu-chahal** [reply]
Hey @amyegan thanks for looking into this.

The app doesn't have any build errors, but the runtime logs give these warnings, I don't know how to solve the issue using these logs:
![ChessBuddy-Vercel-RunTime-Logs](https://github.com/vercel/community/assets/117712672/b0860863-0fd8-4a4c-9dc7-be86f4b41d6f)

The main issue is reloading any page (other than the home page) gives this 404 - Page NOT FOUND ERROR, which the local app doesn't give.

And you're saying you couldn't log in, For me login is working properly, I tried in two browsers MS EDGE and CHROME, can you try once again and share what goes? 

In MS EDGE /games page and its data is loaded properly, but in Chrome, the /games page is loaded but the matches data is not loaded That issue is not related to Vercel, I guess.

But both browsers show 404 NOT FOUND Error on refreshing pages, would appreciate it if you could help with this part.

**Reply by amyegan** [reply]
If you can share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example), that will let me step through the errors to help with debugging.

**Reply by tanu-chahal** [reply, marked as answer]
Hey @amyegan, thanks for the help.

You'll be glad to know that the issue got resolved.

Actually I added a vercel.json in the root directory containing routes (an array of objects) (earlier my project wasn't having any).

After doing that the issue got resolved.

Thanks once again :)

**Reply by amyegan** [reply]
I'm glad you got it working! Thanks for sharing your solution 🙂
